src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DocumentFilePathProvider.cs

```diff
@@ -75,15 +72,23 @@ private string GetGeneratedFilePath(ProjectKey projectKey, string razorFilePath,
         return razorFilePath + projectSuffix + suffix;
     }
 
-    private string GetProjectSuffix(ProjectKey _)
+    private string GetProjectSuffix(ProjectKey projectKey)
     {
         if (!_languageServerFeatureOptions.IncludeProjectKeyInGeneratedFilePath)
         {
             return string.Empty;
         }
 
-        // TODO: Use projectKey to make this dynamic
-        var projectToken = "project";
+        // If there is no project key, we still want to generate something as otherwise the GetRazorFilePath method
+        // would end up unnecessarily overcomplicated
+        if (projectKey.Id is null)
+        {
+            return ".p";
+        }
+
+        using var sha256 = System.Security.Cryptography.SHA256.Create();
+        var crypto = sha256.ComputeHash(Encoding.Unicode.GetBytes(projectKey.Id));
+        var projectToken = Convert.ToBase64String(crypto, 0, 12).Replace('/', '_').Replace('+', '-');
```

Comment:
Might have missed this, is there a reason why we don't want to include `+` in the project token but `-` is OK? Is it because `+` can translate to a space?

Reply:
Yes, correct. `+` is annoying in URLs, and I don't want to risk any bugs.

Reply:
Yeah, Base64 is A-Z, a-z, 0-9, `+` and `/`, and `=` for padding. Probably what we really want it "base 62" but there is nothing in the framework for it. Could probably copy some code from somewhere.